src/connectors/range/connectRange.ts

```diff
@@ -342,10 +420,7 @@ export default function connectRange(renderFn, unmountFn = noop) {
         // can not use setQueryParameters || removeNumericRefinement, because
         // they both keep the old value. This isn't immutable, but it is fine
         // since it's already a copy.
-        stateWithoutDisjunctive.numericRefinements = {
-          ...state.numericRefinements,
-          [attribute]: undefined,
-        };
+        delete stateWithoutDisjunctive.numericRefinements[attribute];
```

Comment:
Any particular reasons for this change?

Reply:
undefined isn't a valid value according to the type, since `x?: y` isn't the same as `x: y | undefined`

Reply:
Right, we might as well change the type in the Helper then?

Reply:
the type in the helper is `numericRefinements: { [facet: string]: SearchParameters.OperatorList }`, do you think that should be `numericRefinements: { [facet: string]: SearchParameters.OperatorList | undefined }`? I feel like we'd need to change a couple more types for that to be consistent. I'd rather keep the delete, it's not a performance footgun anymore nowadays

Reply:
with https://github.com/algolia/algoliasearch-helper-js/pull/801 we can use this for dispose instead (much cleaner IMO)

```js
        return state
          .removeDisjunctiveFacet(attribute)
          .removeNumericRefinement(attribute);
```

Reply:
fixed in https://github.com/algolia/instantsearch.js/pull/4587/commits/93dd0e5edcfc5048c44b94e235a3b0412eff75fa